File: Life/Application/GameService.cs
#region usings

using Domain;
using Infrastructure;

#endregion

namespace Application {
    public class GameService : IGameService {
        public void Save(Game game, IDatabase database) {
            new GameRepository(database).Save(game);
        }

        public Game Load(string name, IDatabase database) {
            return new GameRepository(database).Load(name);
        }
    }
}

File: life-java/src/main/java/ru/agiledays/life/server/application/GameService.java
package ru.agiledays.life.server.application;

import ru.agiledays.life.server.domain.Game;
import ru.agiledays.life.server.infrastructure.Database;

public interface GameService {
    void save(final Game game, final Database database);
    Game load(final String name, final Database database);
}
